src/content/en/fundamentals/instant-and-offline/offline-storage-pwa/index.md

```diff
@@ -0,0 +1,260 @@
+precaching for static assets/application shells)
+
+* Webpack users can directly use the above or
+[offline-plugin](https://github.com/NekR/offline-plugin)
+
+## **How much can I store?**
+
+In Chrome and Opera, your storage is per origin (rather than per API). Both
+storage mechanisms will store data until the browser
+[quota](http://www.html5rocks.com/en/tutorials/offline/quota-research/) is
+reached. Apps can check how much quota they’re using with the [Quota Management
+API](https://www.w3.org/TR/quota-api/). In Firefox, apps can use up to 10% of
+free disk space, but will prompt after 50MB data stored. In mobile Safari, apps
+can use up to 50MB max, whereas desktop Safari allows unlimited storage (and
+prompts after 5MB). IE10+ maxes out at 250MB and prompts at 10MB. PouchDB
+[tracks](https://pouchdb.com/faq.html#data_limits) IDB storage behavior.
+
+## **How can I tell how much storage space my app is using?**
+
+The [Quota Management API](https://www.w3.org/TR/quota-api/) lets you query for
+the size of storage space currently used and how much is available to the
+application. It also enables requesting for more storage if needed.  A newer
+[Storage Quota Estimate
+API](https://www.chromestatus.com/features/5630353511284736) tries to make it
+even easier to discover how much quota an origin is using with support for
+Promises.
+
+## **How does cache eviction work?**
+
+An origin is given an amount of space to do with as it pleases. This free space
+is shared across all forms of origin storage (IndexedDB, Cache API,
+localStorage etc). The amount given isn’t specified and will vary depending on
+device and storage conditions.
+
+When web storage is low, a UA will clear storage to make space available. This
+can harm offline responsiveness so the recently updated
+[Storage](https://storage.spec.whatwg.org/) spec defines "persistent", and
+“best effort” strategies, with “best effort” being the default. “best effort”
+means the storage can be cleared without interrupting the user, but is less
+durable for long-term and/or critical data. IndexedDB and the Cache API both
+fall into the “best effort” category today.
+
+"persistent" storage is not automatically cleared when storage is low - the
+user needs to manually clear this storage (via browser settings). Chrome has
+been experimenting with support for [Persistent
+Storage](/web/updates/2016/06/persistent-storage?hl=en)
+under an origin trial, and the latest news suggests it will be shipping in
+[Chrome
+55](https://groups.google.com/a/chromium.org/d/msg/blink-dev/5Sihi1iAXYc/wnvNDFIPAQAJ).
+
+## **Current and future offline storage work**
+
+If offline storage interests you, the below efforts are worth keeping an eye
+on. One particularly exciting development in progress is native Promises
+support in IndexedDB without the need for a separate library.
+
+![Promise-ized IDB](images/image_1.jpg)
+
+IDB with some await goodness.
```

Comment:
yes, fixed by using the caption tag but let me know if there's a recommended markdown-y way to do it.